ckanext/iati_generator/tests/test_actions_iati_file_update.py

```diff
@@ -0,0 +1,242 @@
+import pytest
+from types import SimpleNamespace
+from ckan.tests import factories
+from ckan import model
+
+from ckanext.iati_generator.models.enums import IATIFileTypes
+from ckanext.iati_generator.models.iati_files import IATIFile
+
+
+@pytest.fixture
+def setup_data():
+    """Base setup with users, organization, dataset and resource, with headers ready."""
+    obj = SimpleNamespace()
+
+    # Users + tokens
+    obj.sysadmin = factories.SysadminWithToken()
+    obj.sysadmin["headers"] = {"Authorization": obj.sysadmin["token"]}
+    obj.user_admin = factories.UserWithToken()
+    obj.user_admin["headers"] = {"Authorization": obj.user_admin["token"]}
+    obj.user_editor = factories.UserWithToken()
+    obj.user_editor["headers"] = {"Authorization": obj.user_editor["token"]}
+    obj.user_member = factories.UserWithToken()
+    obj.user_member["headers"] = {"Authorization": obj.user_member["token"]}
+
+    # Organization + roles
+    obj.org = factories.Organization(
+        users=[
+            {"name": obj.user_admin["name"], "capacity": "admin"},
+            {"name": obj.user_editor["name"], "capacity": "editor"},
+            {"name": obj.user_member["name"], "capacity": "member"},
+        ]
+    )
+
+    # Dataset + resource
+    obj.pkg = factories.Dataset(owner_org=obj.org["id"])
+    obj.res = factories.Resource(
+        package_id=obj.pkg["id"], format="CSV", url_type="upload",
+        url="file.csv", name="file.csv",
+    )
+
+    return obj
+
+
+@pytest.mark.usefixtures("with_plugins", "clean_db")
+class TestIatiFileUpdateAction:
+    """Test suite for iati_file_update action: permissions and correct execution."""
+
+    def _api(self, action):  # helper
+        return f"/api/3/action/{action}"
+
+    def _create_file(self, app, headers, res_id, file_type=IATIFileTypes.ORGANIZATION_MAIN_FILE.name):
```

Comment:
Define a `IATIFileFactory`, like the issue requests

Reply:
https://github.com/okfn/ckanext-iati-generator/pull/55/commits/35acafb7c36d5858dca3c3af2d0a7755614afdc3